### ISSUE TITLE
Inherents are checked with ValidateUnsigned

### ISSUE THREAD

**Issue by nazar-pc** [paritytech/substrate#10256]
I see that `ValidateUnsigned` is not needed for inherents (for instance, `pallet-timestamp` works that way).

However, when pallet implements both `ProvideInherent` and `ValidateUnsigned`, `ValidateUnsigned::validate_unsigned()` is called on inherents for further validation.

This doesn't make a lot of sense to me since there is already `ProvideInherent::check_inherent()` for that purpose.

Maybe `ProvideInherent::is_inherent()` is needed before calling `ValidateUnsigned::validate_unsigned()` and `::pre_dispatch()` on extrinsics that are not signed and depending on response either one or the other kind of validation should be used?

**Comment by thiolliere** [comment]
Yes I think we can improve on the way to declare inherent extrinsics and unsigned transaction to avoid having this trait ValidateUnsigned which needs to take care of both.

Currently ValidateUnsigned should be understood as the trait to validate unsigned extrinsics, so both unsigned transactions and inherents.
here is the implementation https://github.com/paritytech/substrate/blob/cc4bf9134b492d3a11f45c6b1cfa4fdf307e63e2/frame/support/procedural/src/construct_runtime/expand/unsigned.rs#L46-L70
